test/helpers/CMakeLists.txt

```diff
@@ -8,18 +8,18 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
-set(fppFiles pxpotrf.fpp pxheevd.fpp pxhegvx.fpp)
+set(fyppFiles pxpotrf.fypp pxheevd.fypp pxhegvx.fypp)
 
-foreach(infileName IN LISTS fppFiles)
+foreach(infileName IN LISTS fyppFiles)
 
-    string(REGEX REPLACE ".fpp\$" ".f90" outfileName "${infileName}")
+    string(REGEX REPLACE ".fypp\$" ".f90" outfileName "${infileName}")
 
     set(outfile "${CMAKE_CURRENT_BINARY_DIR}/${outfileName}")
     set(infile "${CMAKE_CURRENT_SOURCE_DIR}/${infileName}")
 
     add_custom_command(
         OUTPUT "${outfile}"
-        COMMAND fypp "${infile}" "${outfile}"
+        COMMAND "$CACHE{FYPP_PROGRAM}" "${infile}" "${outfile}"
```

Comment:
Out of curiosity: why are you explicitly targeting the CACHE? In the check I commented above you are using the "generic" access.

If there is no specific reason, I would just go for the "generic" access (which in the end [it might use the cache anyway](https://cmake.org/cmake/help/latest/variable/CACHE.html#cache)) just to not risk strange problems and be sure to use always the same value everywhere.

Reply:
I think I had issues without this, but likely due to an erroneous placement of `add_subdirectory`. Just re-tested without `CACHE` now, and it works. Fixed.